### ISSUE TITLE
Conda env update does not update pip packages

### ISSUE THREAD

**Issue by mmrahn** [conda/conda#7774]
## Current Behavior
`conda env update` does not update pip packages if pip has version 18.0.

### Steps to Reproduce

```
conda create -n test-env python=3.6
activate test-env
pip install cerberus==1.1
```
Content of `environment.yml`:
```
name: test-env
dependencies:
  - pip:
    - cerberus==1.2
```
If I run `conda env update` with `pip 10.0.1` (which is already installed), I get the correct result:
```
$ conda env update
Solving environment: ...working... done
Preparing transaction: ...working... done
Verifying transaction: ...working... done
Executing transaction: ...working... done
Looking in indexes: https://pypi.org/simple, http://10.2.0.11:8044
Collecting cerberus==1.2 (from -r C:\Users\monar\Git\test_repository\condaenv.n2smo_hm.requirements.txt (line 1))
Installing collected packages: cerberus
  Found existing installation: Cerberus 1.1
    Uninstalling Cerberus-1.1:
      Successfully uninstalled Cerberus-1.1
Successfully installed cerberus-1.2
```
Also after downgrading `cerberus` with `pip install cerberus==1.1`, `conda env update` yields the same correct output.

However, after updating pip to 18.0 via `python -m pip install --upgrade pip`, the `cerberus` package is not updated anymore from the `environment.yml`:
```
pip install cerberus==1.1
conda env update
```
yields
```
$ conda env update
Solving environment: ...working... done
Preparing transaction: ...working... done
Verifying transaction: ...working... done
Executing transaction: ...working... done
#
# To activate this environment, use
#
#     $ conda activate test-env
#
# To deactivate an active environment, use
#
#     $ conda deactivate
```
even though `cerberus` is still at version `1.1`: `pip show cerberus` yields 
```
Name: Cerberus
Version: 1.1
Summary: Lightweight, extensible schema and data validation tool for Python dictionaries.
Home-page: http://github.com/nicolaiarocci/cerberus
Author: Nicola Iarocci
Author-email: nicola@nicolaiarocci.com
License: ISC
Location: c:\users\monar\miniconda3\envs\test-env\lib\site-packages
Requires:
Required-by:
(test-env)
```
(Note that downgrading `pip` via `python -m pip install pip==10.0.1` weirdly enough does not make `conda env update` go back to the correct behavior.)

## Expected Behavior

The correct update of `cerberus`, as it used to be with the older pip version.

## Environment Information
<details open><summary><code>`conda info`</code></summary><p>

```

     active environment : test-env
    active env location : C:\Users\monar\Miniconda3\envs\test-env
            shell level : 1
       user config file : C:\Users\monar\.condarc
 populated config files : C:\Users\monar\.condarc
          conda version : 4.5.11
    conda-build version : not installed
         python version : 3.6.3.final.0
       base environment : C:\Users\monar\Miniconda3  (writable)
           channel URLs : https://repo.anaconda.com/pkgs/main/win-64
                          https://repo.anaconda.com/pkgs/main/noarch
                          https://repo.anaconda.com/pkgs/free/win-64
                          https://repo.anaconda.com/pkgs/free/noarch
                          https://repo.anaconda.com/pkgs/r/win-64
                          https://repo.anaconda.com/pkgs/r/noarch
                          https://repo.anaconda.com/pkgs/pro/win-64
                          https://repo.anaconda.com/pkgs/pro/noarch
                          https://repo.anaconda.com/pkgs/msys2/win-64
                          https://repo.anaconda.com/pkgs/msys2/noarch
          package cache : C:\Users\monar\Miniconda3\pkgs
                          C:\Users\monar\AppData\Local\conda\conda\pkgs
       envs directories : C:\Users\monar\Miniconda3\envs
                          C:\Users\monar\AppData\Local\conda\conda\envs
                          C:\Users\monar\.conda\envs
               platform : win-64
             user-agent : conda/4.5.11 requests/2.18.4 CPython/3.6.3 Windows/10 Windows/10.0.17134
          administrator : False
             netrc file : None
           offline mode : False
```
</p></details>

<details open><summary><code>`conda config --show-sources`</code></summary><p>

```
==> C:\Users\monar\.condarc <==
ssl_verify: True
channels:
  - defaults
```
</p></details>

<details><summary><code>`conda list --show-channel-urls`</code></summary><p>

```
# packages in environment at C:\Users\monar\Miniconda3\envs\test-env:
#
# Name                    Version                   Build  Channel
Cerberus                  1.1                       <pip>
certifi                   2018.8.24                py36_1    defaults
pip                       10.0.1                    <pip>
pip                       10.0.1                   py36_0    defaults
python                    3.6.6                hea74fb7_0    defaults
setuptools                40.2.0                   py36_0    defaults
vc                        14                   h0510ff6_3    defaults
vs2015_runtime            14.0.25123                    3    defaults
wheel                     0.31.1                   py36_0    defaults
wincertstore              0.2              py36h7fe50ca_0    defaults
```
</p></details>
I'm using Windows 10.

**Comment by scholtalbers** [comment]
I just encountered this issue on 4.6.7, but it seems to be resolved (at least in combination with `-f environment.yml`) in conda 4.6.10

**Comment by dhimmel** [comment]
I think this issue has been fixed in https://github.com/conda/conda/pull/8542 by @jpgill86. Not sure if it has been included in a release yet.

**Comment by jpgill86** [comment]
The change in #8542 is included in release 4.6.13 and later.

**Comment by dongbohu** [comment]
Conda updated to `4.6.14`.
